src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.Unix.cs

```diff
@@ -1,111 +1,141 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 
 namespace System.Threading
 {
-    //
-    // Portable implementation of ThreadPool
-    //
-
-    internal sealed partial class CompleteWaitThreadPoolWorkItem : IThreadPoolWorkItem
-    {
-        void IThreadPoolWorkItem.Execute() => PortableThreadPool.CompleteWait(_registeredWaitHandle, _timedOut);
-    }
-
     public static partial class ThreadPool
     {
-        // Indicates whether the thread pool should yield the thread from the dispatch loop to the runtime periodically so that
-        // the runtime may use the thread for processing other work
-#if !(TARGET_BROWSER && FEATURE_WASM_THREADS)
-        internal static bool YieldFromDispatchLoop => false;
-#endif
-
 #if NATIVEAOT
         private const bool IsWorkerTrackingEnabledInConfig = false;
 #else
         private static readonly bool IsWorkerTrackingEnabledInConfig =
             AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.EnableWorkerTracking", false);
 #endif
 
-        // Threadpool specific initialization of a new thread. Used by OS-provided threadpools. No-op for portable threadpool.
-        internal static void InitializeForThreadPoolThread() { }
+#if CORECLR || !(TARGET_BROWSER && FEATURE_WASM_THREADS)
+        // Indicates whether the thread pool should yield the thread from the dispatch loop to the runtime periodically so that
+        // the runtime may use the thread for processing other work.
+        internal static bool YieldFromDispatchLoop => false;
+#endif
+
+#if !CORECLR
+        private static bool EnsureConfigInitializedCore() => true;
+#endif
 
-#pragma warning disable IDE0060
-        internal static bool CanSetMinIOCompletionThreads(int ioCompletionThreads) => false;
-        internal static bool CanSetMaxIOCompletionThreads(int ioCompletionThreads) => false;
-#pragma warning restore IDE0060
+        internal static bool EnsureConfigInitialized() => EnsureConfigInitializedCore();
```

Comment:
```suggestion
```
There does not seem to be a reason for having EnsureConfigInitialized split into two methods.